search/tests/tests.py

```diff
@@ -0,0 +1,1442 @@
+
+    def test_excerpt_back(self):
+        """ test that we process correctly when match is at the end of the excerpt """
+        test_result = {
+            "content": {
+                "notes": "Match upon last word - Dog",
+            }
+        }
+        srp = SearchResultProcessor(test_result, "dog")
+        self.assertEqual(srp.excerpt, "Match upon last word - <b>Dog</b>")
+
+        test_result = {
+            "content": {
+                "notes": (
+                    "The long and winding road "
+                    "That leads to your door "
+                    "Will never disappear "
+                    "I've seen that road before "
+                    "It always leads me here "
+                    "Lead me to you door "
+                    "The wild and windy night "
+                    "That the rain washed away "
+                    "Has left a pool of tears "
+                    "Crying for the day "
+                    "Why leave me standing here "
+                    "Let me know the way "
+                    "Many times I've been alone "
+                    "And many times I've cried "
+                    "Any way you'll never know "
+                    "The many ways I've tried "
+                    "But still they lead me back "
+                    "To the long winding road "
+                    "You left me standing here "
+                    "A long long time ago "
+                    "Don't leave me waiting here "
+                    "Lead me to your door "
+                    "But still they lead me back "
+                    "To the long winding road "
+                    "You left me standing here "
+                    "A long long time ago "
+                    "Don't leave me waiting here "
+                    "Lead me to your door "
+                    "Yeah, yeah, yeah, yeah "
+                    "Match upon last word - Dog"
+                ),
+            }
+        }
+        srp = SearchResultProcessor(test_result, "dog")
+        self.assertEqual(srp.excerpt[-33:], "Match upon last word - <b>Dog</b>")
+
+
+class OverrideSearchResultProcessor(SearchResultProcessor):
+    """
+    Override the SearchResultProcessor so that we get the additional (inferred) properties
+    and can identify results that should be removed due to access restriction
+    """
+    # pylint: disable=no-self-use
+    @property
+    def additional_property(self):
+        """ additional property that should appear within processed results """
+        return "Should have an extra value"
+
+    @property
+    def url(self):
+        """
+        Property to display the url for the given location, useful for allowing navigation
+        """
+        if "course" not in self._results_fields or "id" not in self._results_fields:
+            raise ValueError("expect this error when not providing a course and/or id")
+
+        return u"/courses/{course_id}/jump_to/{location}".format(
```

Comment:
url does not appear to be present in latest version - it has been removed as far as I can tell.

> On Jan 27, 2015, at 2:54 PM, chrisndodge <notifications@github.com> wrote:
> 
> In search/tests/tests.py <https://github.com/edx/edx-search/pull/1#discussion_r23637253>:
> 
> > +    """
> > +    # pylint: disable=no-self-use
> > +    @property
> > +    def additional_property(self):
> > +        """ additional property that should appear within processed results """
> > +        return "Should have an extra value"
> > +
> > +    @property
> > +    def url(self):
> > +        """
> > +        Property to display the url for the given location, useful for allowing navigation
> > +        """
> > +        if "course" not in self._results_fields or "id" not in self._results_fields:
> > +            raise ValueError("expect this error when not providing a course and/or id")
> > +
> > +        return u"/courses/{course_id}/jump_to/{location}".format(
> I think we discussed this as a encapsulation problem in the past. I believe you said that the LMS will create a specialized subclass and overload this method, right?
> 
> If this is meant to be an abstract property, can you simply just raise a NotImplementedError()?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/edx/edx-search/pull/1/files#r23637253>.
>